### DISCUSSION TITLE
Advice for loading CSV data for a ScatterplotLayer alongside TileLayer Image data

### DISCUSSION THREAD

**Discussion by cornhundred** [visgl/deck.gl#5519] [Q&A] [answered]
We have a large image dataset that we are visualizing using the TileLayer, but we would also like to overlay a scatterplot on top of the TileLayer. However, we have too many points to show all at once in the ScatterplotLayer and would probably only want to show points when zoomed in. What would be the best approach for this? 

I am thinking I could save CSVs corresponding to the image tiles (at their highest resolution) and then load them when we zoom in. Should I load them simultaneously to loading the tiles? Would the streaming support for loading this scatterplot data work?

**Comment by kylebarron** [comment, marked as answer]
Why don't you tile the points as well? Consider https://github.com/mapbox/tippecanoe to build vector tiles that are served with your images.

**Reply by cornhundred** [reply]
Thanks @kylebarron, I'll check it out

**Reply by cornhundred** [reply]
@kylebarron, would I need to run a vector tile server to host the data or use Mapbox to host? Is it possible to have GitHub host the tile files (e.g. mbtiles file) as is done with the TileLayer example?

Update: I see in your [blog post](https://kylebarron.dev/blog/all-transit) and [repo](https://github.com/kylebarron/all-transit) that you use mbtileserver to host the tiles or exported them as files to b hosted on AWS S3. I'l see if I can export my MVT tiles as files and then have them be read from GitHub.
